tests/ToStringTest.php

```diff
@@ -208,6 +208,16 @@ public function __toString() {
 
                     echo implode(":", [new Bar()]);',
             ],
+            'resourceOrFalseToString' => [
+                'code' => '<?php
+                    $a = fopen("php://memory", "r");
+                    if (rand(0, 1)) {
+                        $a = [];
+                    }
+
+                    // b = "Array"|""|"Resource id #1"
+                    $b = (string) $a;',
+            ],
         ];
```

Comment:
previously considered failing due to the assumpton plsam made that `[]` can't be cast to string.